client/components/questionnaire2.js

```diff
@@ -1,61 +1,109 @@
 import React, {Component} from 'react'
+import SubQuestions from './subcategories-questions'
 
 class Questionnaire2 extends Component {
   constructor(props) {
     super(props)
     this.handleSubmit = this.handleSubmit.bind(this)
     this.handleChange = this.handleChange.bind(this)
     this.state = {
-      food: '',
-      shopping: '',
-      publicTransit: '',
-      personalTransport: ''
+      entertainment: {
+        name: 'entertainment',
+        overallMonthly: '',
+        subcategories: []
+      },
+      publicTransit: {
+        name: 'publicTransit',
+        overallMonthly: '',
+        subcategories: []
+      },
+      personalTransport: {
+        name: 'personalTransport',
+        overallMonthly: '',
+        subcategories: []
+      }
     }
   }
 
   handleChange = event => {
-    this.setState({...this.state, [event.target.name]: event.target.value})
+    let stateName = event.target.name // i.e. Public Transportation
+    let newStateName = '' //want to get publicTransportation
+    for (let i = 0; i < stateName.length; i++) {
+      let currentLetter = stateName[i]
+      if (i === 0) {
+        currentLetter = currentLetter.toLowerCase()
+      }
+      if (currentLetter !== ' ') {
+        newStateName += currentLetter
+      }
+    }
+    this.setState({
+      ...this.state,
+      [newStateName]: {
+        name: event.target.name,
+        overallMonthly: event.target.value
+      }
+    })
   }
 
   handleSubmit = event => {
     event.preventDefault()
-    this.props.history.push('/questionnaire-page3')
+    //---this will check to see if the subcategories add up to the overall total---
+    // for (let keys in this.state){
+    //   let total = Number(keys.overallMonthly)
+    //   let check = 0;
+    //   keys.subcategories.map(current => {
+    //     if (current.monthly.length > 0){
+    //       current.monthly = Number(current.monthly)
+    //       check += current.monthly
+    //     }
+    //     }
+    //   )
+    //   if (total !== check){
+    //     alert(`Your overall monthly amount in ${keys.name} does not match the added total of the subcategories in ${keys.name}. Please make changes.`)
+    //   }
+    // }
     console.log(this.state)
```

Comment:
Unsure if you are removing or refactoring this?

Reply:
I left that in because it's supposed to be the validation that checks to see if the subcategories add up to the overall monthly amount. However, it's commented out currently because the subcategories has an issue where it only stores 1 digit to the state.